### ISSUE TITLE
[BUG] NameError on deprecation warning in 58.4.0

### ISSUE THREAD

**Issue by austinweisgrau** [pypa/setuptools#2846]
### setuptools version

58.4.0

### Python version

Python 3.6

### OS

Ubuntu 18.04

### Description

A version conflict resulted in setuptools attempting to raise a class that no longer exists in 58.4.0, resulting in a crash.

### Expected behavior

The class name should be called appropriately so that a NameError isn't raised

### How to Reproduce

In particular, this happened when trying to launch gunicorn.

### Output

```console
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: Traceback (most recent call last):
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 573, in _build_master
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     ws.require(__requires__)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 891, in require
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     needed = self.resolve(parse_requirements(requirements))
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 782, in resolve
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     raise VersionConflict(dist, req).with_context(dependent_req)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: pkg_resources.VersionConflict: (gunicorn 19.10.0 (/home/ubuntu/.local/lib/python3.6/site-packages), Requirement.parse('gunicorn==19.7.1'))
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: During handling of the above exception, another exception occurred:
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: Traceback (most recent call last):
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 114, in parse_version
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     return packaging.version.Version(v)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/_vendor/packaging/version.py", line 277, in __init__
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     raise InvalidVersion("Invalid version: '{0}'".format(version))
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: pkg_resources.extern.packaging.version.InvalidVersion: Invalid version: '0.18ubuntu0.18.04.1'
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: During handling of the above exception, another exception occurred:
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: Traceback (most recent call last):
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/usr/bin/gunicorn3", line 6, in <module>
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     from pkg_resources import load_entry_point
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3247, in <module>
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     @_call_aside
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3231, in _call_aside
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     f(*args, **kwargs)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3260, in _initialize_master_working_set
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     working_set = WorkingSet._build_master()
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 575, in _build_master
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     return cls._build_from_requirements(__requires__)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 588, in _build_from_requirements
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     dists = ws.resolve(reqs, Environment())
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 975, in __init__
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     self.scan(search_path)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1008, in scan
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     self.add(dist)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1028, in add
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     dists.sort(key=operator.attrgetter('hashcmp'), reverse=True)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2604, in hashcmp
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     self.parsed_version,
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2651, in parsed_version
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     self._parsed_version = parse_version(self.version)
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:   File "/home/ubuntu/.local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 119, in parse_version
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]:     PkgResourcesDeprecationWarning,
Nov 02 20:03:22 ip-172-31-23-12 gunicorn3[9718]: NameError: name 'PkgResourcesDeprecationWarning' is not defined
```

### Code of Conduct

- [X] I agree to follow the PSF Code of Conduct

**Comment by jaraco** [comment]
Thanks for the report. Yes, that definitely sounds like a bug.

**Comment by jaraco** [comment]
That class does still exist. I think the issue is that the warning gets raised before the parser gets to processing the definition of the warning class in the module.

**Comment by jaraco** [comment]
Presumed fix releasing as v58.5.1
